.buildkite/scripts/build_component.sh

```diff
@@ -41,37 +41,11 @@ else
 fi
 source results/last_build.env
 
-# TODO (SM): The 0.59.0 hab cli that we rely on for x86_64-linux builds 
-# doesn't emit pkg_target. Until we've sufficiently bootstrapped ourselves
-# we need to set it. This can be removed when studio-ci-common pulls 0.63.0 
-# or newer. This is safe to do because the x86_64-linux-kernel2 builds will
-# already have this value set.
-: "${pkg_target:=x86_64-linux}"
-
-
-# TODO: after 0.79.0 we can reenable this. We are explicitly using curl to upload
-# due to this bug: https://github.com/habitat-sh/builder/issues/940
-# echo "--- :habicat: Uploading ${pkg_ident} to Builder in the '${channel}' channel"
-# ${hab_binary} pkg upload \
-#     --channel="${channel}" \
-#     --auth="${HAB_AUTH_TOKEN}" \
-#     "results/${pkg_artifact}"
-#
-# ${hab_binary} pkg promote \
-#     --auth="${HAB_AUTH_TOKEN}" \
-#     "${pkg_ident}" "${channel}" "${pkg_target}"
-
-echo "--- :partyparrot: Manually uploading '${pkg_ident:?}' (${pkg_target}) to Builder"
-curl --request POST \
-     --header "Content-Type: application/octet-stream" \
-     --header "Authorization: Bearer $HAB_AUTH_TOKEN" \
-     --data-binary "@results/${pkg_artifact:?}" \
-     --fail \
-     --verbose \
-    "https://bldr.habitat.sh/v1/depot/pkgs/${pkg_ident}?checksum=${pkg_blake2bsum:?}&target=${pkg_target}"
-
-promote "${pkg_ident}" "${pkg_target}" "${channel}"
-set_target_metadata "${pkg_ident}" "${pkg_target}"
+echo "--- :habicat: Uploading ${pkg_ident} to Builder in the '${channel}' channel"
+${hab_binary} pkg upload \
+    --channel="${channel}" \
+    --auth="${HAB_AUTH_TOKEN}" \
+    "results/${pkg_artifact}"
```

Comment:
We should double-check that this behaves properly with respect to package target before getting this merged. We can do a "fake release", as detailed in https://github.com/habitat-sh/habitat/pull/6510.